src/protocol/communication.rs

```diff
@@ -153,10 +151,10 @@ fn receive_payload(
         }
 
         bytes_read += size;
-        if bytes_read == length as usize {
+        if bytes_read == length {
             match serde_json::from_slice(&buffer) {
                 Ok(v) => {
-                    log::debug!("received {:?} from {}", v, stream.peer_addr()?);
+                    log::debug!("received from {} {:?}", stream.peer_addr()?, v);
```

Comment:
Same idea about delimiting fields here.

I get that the more salient bit of information when looking at a stream of events is the source address (though if you have multiple active connections when debugging, you're already going to be dealing with a ton of information), but when you're only dealing with a single peer, which is more common when troubleshooting, seeing the message first may be more helpful because that's the part that is actually interesting.

If, as with the previous logging change, you want to keep this order of information, but also make it flow naturally in English, consider `"received message from {}: {:?}"`